src/Nethermind/Nethermind.TxPool/Filters/SizeTxFilter.cs

```diff
@@ -0,0 +1,31 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core;
+using Nethermind.Logging;
+
+namespace Nethermind.TxPool.Filters;
+
+/// <summary>
+/// Ignores transactions that exceed configured max transaction size limit.
+/// </summary>
+internal sealed class SizeTxFilter(ITxPoolConfig txPoolConfig, ILogger logger) : IIncomingTxFilter
+{
+    private readonly long _configuredMaxTxSize = txPoolConfig.MaxTxSize ?? long.MaxValue;
+
+    public AcceptTxResult Accept(Transaction tx, ref TxFilteringState state, TxHandlingOptions txHandlingOptions)
+    {
+        if (tx.SupportsBlobs)
+        {
+            return AcceptTxResult.Accepted;
+        }
```

Comment:
I was sure it can't, but looking at the spec and code - there is no restriction on it. Just can't create a contract (`to` can't be `null`).

Geth is limiting non-blob part to 1MB https://github.com/ethereum/go-ethereum/blob/master/core/txpool/blobpool/blobpool.go#L63-L68

I think I will add here a limit of 8x standard limit (8x128KiB).

Great catch!